pkg/parquetquery/iters.go

```diff
@@ -600,9 +600,9 @@ func (c *SyncIterator) SeekTo(to RowNumber, definitionLevel int) (*IteratorResul
 	}
 }
 
-func (c *SyncIterator) popRowGroup() (pq.RowGroup, RowNumber, RowNumber) {
+func (c *SyncIterator) popRowGroup() (pq.RowGroup, *RowNumber, *RowNumber) {
```

Comment:
Oh meant to add that maybe the ideal solution is to change [RowNumber](https://github.com/grafana/tempo/blob/main/pkg/parquetquery/iters.go#L33) to a pointer.

> slice instead of an array

Array should be less overhead because it doesn't allocate the slice struct nor the length.

Reply:
I could not repro popRowGroup's impact so i've removed it.